src/Detector/DngDetector.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Selective\ImageType\Detector;
+
+use Selective\ImageType\ImageType;
+use SplFileObject;
+
+/**
+ * Detector.
+ */
+final class DngDetector implements DetectorInterface
+{
+    /**
+     * DNG identification.
+     *
+     * @param SplFileObject $file The image file
+     *
+     * @return ImageType|null The image type
+     */
+    public function detect(SplFileObject $file): ?ImageType
+    {
+        $file->rewind();
+        $bytes = $file->fread(2);
+
+        // TIFF header
+        if ($bytes !== 'II' && $bytes !== 'MM') {
+            return null;
+        }
+
+        $file->fread(6);
+        $bytes = $file->fread(12) ?: "";
+
+        return ((strpos($bytes, "\x04") !== false || strpos($bytes, "\x02") !== false) && strpos($bytes, "\x01") && substr_count($bytes, "\0") >= 2) ? new ImageType(ImageType::DNG) : null;
```

Comment:
The `DNG` image detection approach is as follows:
- Reading first `2` bytes firstly to make sure that it's the `TIFF` header.
- Reading the next `12` bytes that is about the `IFD0`.
- Using the condition to detect that it will contain `\x04` or `\x02` byte and it should contain `\x01` byte.
- Using the condtition to detect that it will contain 2 `\0` bytes at least.
- This detection is to check whether the DNG version is `1,2,0,0` or `1,4,0,0`.